### PR TITLE
provide hyperlinks to `mapping` and `enumeration`

### PR DESCRIPTION
**Pull request by MichaelSodeke**
Beginners to PTES and bug bounties, like me, maybe confused by the word **mapping** and **enumeration**. By providing links for each word to the appropriate sections of the WSTG that cover them, the reader does not have to go down a 'search-engine-rabbit-hole' to find information about the two.

Take care.

### CODE REVIEW

File: document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Below are some points of interests for all requests and responses. Within the re
 
 - Identify where GETs are used and where POSTs are used.
 - Identify all parameters used in a POST request (these are in the body of the request).
-- Within the POST request, pay special attention to any hidden parameters. When a POST is sent all the form fields (including hidden parameters) will be sent in the body of the HTTP message to the application. These typically aren't seen unless a proxy or view the HTML source code is used. In addition, the next page shown, its data, and the level of access can all be different depending on the value of the hidden parameter(s).
+- Within the POST request, pay special attention to any hidden parameters. When a POST is sent all the form fields (including hidden parameters) will be sent in the body of the HTTP message to the application. These typically aren't seen `unless a proxy or view the HTML source code is used`. In addition, the next page shown, its data, and the level of access can all be different depending on the value of the hidden parameter(s).
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:31]: This isn’t code. There doesn’t seem to be any need to quote or highlight this part of the text.

File: document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md [hunk 2]
```diff
@@ -6,15 +6,15 @@
 
 ## Summary
 
-Enumerating the application and its attack surface is a key precursor before any thorough testing can be undertaken, as it allows the tester to identify likely areas of weakness. This section aims to help identify and map out areas within the application that should be investigated once enumeration and mapping have been completed.
+Enumerating the application and its attack surface is a key precursor before any thorough testing can be undertaken, as it allows the tester to identify likely areas of weakness. This section aims to help identify and map out areas within the application that should be investigated once [mapping ``4.1.2``](https://github.com/OWASP/wstg/blob/master/document/4-Web_Application_Security_Testing/01-Information_Gathering/02-Fingerprint_Web_Server.md) and [enumeration ``4.1.4``](https://github.com/OWASP/wstg/blob/master/document/4-Web_Application_Security_Testing/01-Information_Gathering/04-Enumerate_Applications_on_Webserver.md) have been completed.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:9]: Links should be relative and the anchor text needs to be the actual name of the content in order for the pdf version to link properly.

File: document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md [hunk 3]
```diff
@@ -6,15 +6,15 @@
...
 
 ## How to Test
 
-Before any testing begins, the tester should always get a good understanding of the application and how the user and browser communicates with it. As the tester walks through the application, they should pay attention to all HTTP requests as well as every parameter and form field that is passed to the application. They should pay special attention to when GET requests are used and when POST requests are used to pass parameters to the application. In addition, they also need to pay attention to when other methods for RESTful services are used.
+Before any testing begins, the tester should always get a good understanding of the application and how the user and browser communicates with it. As the tester walks through the application, they should pay attention to all HTTP requests as well as every parameter and form field that is passed to the application. They should pay special attention to when GET requests are used and when POST requests are used to pass parameters to the application. In addition, they also need to pay attention to when other methods for REST-ful services are used.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:17]: I only looked quickly but RESTful seems to be used twice within the doc while the hyphenated variant doesn’t appear at all.

I’d suggest rolling this change back. If we need to prefer one over the other it should be changed consistently throughout and the text linter should be updated with a related rule. (Also it’s off topic or out of scope for this PR.)

**Comment by kingthorin** [comment]
> MichaelSodeke closed this 1 hour ago
> MichaelSodeke MichaelSodeke deleted the patch-4 branch 1 hour ago 

Why they were easily addressable comments?

File: document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md (hunk 3 above)
Comment (kingthorin) [reply document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:17]: Just in-case this comes up again in the future it seems people prefer RESTful:
https://twitter.com/kingthorin_rm/status/1480611497676001288

image

Comment (MichaelSodeke) [reply document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:17]: No problem.

Comment (kingthorin) [reply document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:17]: That wasn't meant as a "told ya so" kind of thing. I just figured it was worth asking, and I was just following up here for the team (or anyone that might care). Sorry if it came across the wrong way.

Comment (MichaelSodeke) [reply document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:17]: I almost forgot. Is there a way I can join OWASP here on GitHub? Are there any requirements?

I know I'm still a Noob in network security, but I would like to help with proofreading and code reviewing for OWASP. Is this OK with you?

Take care.

Comment (kingthorin) [reply document/4-Web_Application_Security_Testing/01-Information_Gathering/06-Identify_Application_Entry_Points.md:17]: To join OWASP check owasp.org/membership

You don’t need to be a member to contribute to projects though, feel free to jump in anywhere you see something that catches your eye/interest.

**Pull request closed without merging**